### DISCUSSION TITLE
Hope to support more mouse association features

### DISCUSSION THREAD

**Discussion by chenbihao** [mulaRahul/keyviz#54] [Ideas]
I like this project very much, make a suggestion

Can add a drag-and-drop display window function, instead of fixed in the corner;

At the same time, I hope that the mouse buttons can be displayed (such as left and right keys, rolling wheel sliding, side keys, etc.), and display at the same time as the keyboard button

**Comment by mulaRahul** [comment]
Adding 🖱️ mouse support is already [planned](https://github.com/mulaRahul/keyviz/discussions/36#discussion-4327240)! About the _drag-n-drop_ positioning of the visualization, flutter (the framework used to build the application) can only show one window. That's why it's a bit hard to implement the same 😅. 

I've thought of separating the settings window and the visualization. So that it's easier to view the position, etc. changes on the visualization while the settings window is open. Also, [more alignment options](https://github.com/mulaRahul/keyviz/discussions/14#discussioncomment-3493004) will be added. If everything works out, then it'll be great.

**Reply by chenbihao** [reply]
About the drag-n-drop positioning of the visualization, i mean like Suspended Lyrics of music appcaltion，you can drag the position of the lyrics at any time without going to the settings interface to adjust

**Reply by mulaRahul** [reply]
Dragging and re-positioning a normal window would be easy, but it's not possible here because the visualization window covers half of the screen. The same is done so that you can align the visualization to the bottom left, center and right.

https://github.com/mulaRahul/keyviz/blob/a4f297121ef3c86921a4c8692785b7fdd75d220e/lib/main.dart#L44

That's why **mouse events are ignored** else you would not be able to click on the applications/windows beneath it.

https://github.com/mulaRahul/keyviz/blob/a4f297121ef3c86921a4c8692785b7fdd75d220e/lib/main.dart#L49

**Comment by mulaRahul** [comment]
@chenbihao, mouse visualization added with [v2.0.0a](https://github.com/mulaRahul/keyviz/releases/tag/v2.0.0a).

**Reply by chenbihao** [reply]
so cool ,

there is still a minor issue. When I press the side button and middle key, the visualization does not disappear.

**Reply by mulaRahul** [reply]
probably bug, will look into the same
